app/main.ipynb

```diff
@@ -56,33 +57,59 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 5,
-   "outputs": [],
    "source": [
-    "# write your code here (for Python default syntax)"
+    "array = [0] * 1000\n",
+    "\n",
+    "sys.getsizeof(array)"
    ],
    "metadata": {
     "collapsed": false,
     "ExecuteTime": {
-     "end_time": "2023-07-03T12:02:41.673421Z",
-     "start_time": "2023-07-03T12:02:41.666277Z"
+     "end_time": "2024-11-10T02:39:20.405539Z",
+     "start_time": "2024-11-10T02:39:20.401056Z"
     }
-   }
+   },
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "8056"
+      ]
+     },
+     "execution_count": 7,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "execution_count": 7
   },
   {
    "cell_type": "code",
-   "execution_count": 4,
-   "outputs": [],
    "source": [
-    "# write your code here (for numpy syntax)"
+    "np_array = np.array(array, dtype=np.int8)\n",
+    "\n",
+    "sys.getsizeof(np_array)"
```

Comment:
The NumPy array is created from the existing Python list. To ensure minimal memory usage, you might want to directly create the array using `np.zeros` with `dtype=np.int8` instead of converting from a list.